src/services/api/api.ts

```diff
@@ -0,0 +1,39 @@
+import { type BooksApiResponse } from "../../types/index";
+import { type Book } from "../../types/book.types";
+
+const API_URL = "https://www.googleapis.com/books/v1/volumes";
+const MAX_ITEMS = 14;
+// промисы обещают вернуть заданные типы
+export const fetchBooks = async (
+  query: string,
+  startIndex: number = 0,
+  additionalFilters: string = ""
+): Promise<BooksApiResponse> => {
+  try {
+    // применяю Utility Types
+    const params: Record<string, string> = {
+      q: query,
+      startIndex: startIndex.toString(),
+      maxResults: MAX_ITEMS.toString(),
+    };
+    if (additionalFilters) {
+      params.filter = additionalFilters;
+    }
```

Comment:
Тут можно покрасивее

```suggestion
    const params: Record<string, string> = {
      q: query,
      startIndex: startIndex.toString(),
      maxResults: MAX_ITEMS.toString(),
      ...(additionalFilters && { filter: additionalFilters }),
    };
```